packages/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashMainActivity.ts

```diff
@@ -0,0 +1,117 @@
+import { AndroidConfig, ConfigPlugin, withMainActivity } from '@expo/config-plugins';
+import { mergeContents, removeContents } from '@expo/config-plugins/build/utils/generateCode';
+import { ExpoConfig } from '@expo/config-types';
+import Debug from 'debug';
+
+import { getAndroidSplashConfig } from './getAndroidSplashConfig';
+
+const debug = Debug('expo:prebuild-config:expo-splash-screen:android:mainActivity');
+
+// DO NOT CHANGE
+const SHOW_SPLASH_ID = 'expo-splash-screen-mainActivity-onCreate-show-splash';
+
+export const withAndroidSplashMainActivity: ConfigPlugin = config => {
+  return withMainActivity(config, config => {
+    config.modResults.contents = setSplashScreenMainActivity(
+      config,
+      config.modResults.contents,
+      config.modResults.language
+    );
+    return config;
+  });
+};
+
+export function setSplashScreenMainActivity(
+  config: Pick<ExpoConfig, 'android' | 'androidStatusBar' | 'userInterfaceStyle'>,
+  mainActivity: string,
+  language: 'java' | 'kt'
+): string {
+  debug(`Modify with language: "${language}"`);
+  const splashConfig = getAndroidSplashConfig(config);
+
+  if (!splashConfig) {
+    // Remove our generated code safely...
+    const mod = removeContents({
+      src: mainActivity,
+      tag: SHOW_SPLASH_ID,
+    });
+
+    mainActivity = mod.contents;
+    if (mod.didClear) {
+      debug('Removed SplashScreen.show()');
+    }
+    return mainActivity;
+  }
+  // TODO: Translucent is weird
+  const statusBarTranslucent = !!config.androidStatusBar?.translucent;
+
+  const { resizeMode } = splashConfig;
+  const isJava = language === 'java';
+  const LE = isJava ? ';' : '';
+
+  mainActivity = AndroidConfig.UserInterfaceStyle.addJavaImports(
+    mainActivity,
+    [
+      'expo.modules.splashscreen.SplashScreen',
+      'expo.modules.splashscreen.SplashScreenImageResizeMode',
+      'android.os.Bundle',
+    ],
+    isJava
+  );
+
+  if (!mainActivity.match(/(?<=^.*super\.onCreate.*$)/m)) {
+    const onCreateBlock = isJava
+      ? [
+          '    @Override',
+          '    protected void onCreate(Bundle savedInstanceState) {',
+          '      super.onCreate(savedInstanceState);',
+          '    }',
+        ]
+      : [
+          '    override fun onCreate(savedInstanceState: Bundle?) {',
+          '      super.onCreate(savedInstanceState)',
+          '    }',
+        ];
+
+    mainActivity = mergeContents({
+      src: mainActivity,
+      // insert just below super.onCreate
+      anchor: isJava ? /(?<=public class .* extends .* {.*$)/m : /(?<=class .* : .* {.*$)/m,
```

Comment:
nit: sometimes if the line is too long, people may break to multiple lines
e.g.

https://github.com/facebook/react-native/blob/27651720b40cab564a0cbd41be56a02584e0c73a/ReactAndroid/src/main/java/com/facebook/react/views/progressbar/ReactProgressBarViewManager.java#L30-L32

maybe we could use regex like this to support multiple line matching if not feel too complicated
```suggestion
      anchor: isJava ? /(?<=public\s+class\s+.*\s+extends\s+.*\s+{.*$)/ms : /(?<=class\s+.*\s+:\s+.*\s+{.*$)/ms, 
```

or, if we are targeting our templates only, please just ignore this.